lnwallet/channel.go

```diff
@@ -1867,36 +1869,57 @@ func (lc *LightningChannel) restorePendingRemoteUpdates(
 			return err
 		}
 
+		logIdx := payDesc.LogIndex
+
 		// Sanity check that we are not restoring a remote log update
 		// that we haven't received a sig for.
-		if payDesc.LogIndex >= lc.remoteUpdateLog.logIndex {
+		if logIdx >= lc.remoteUpdateLog.logIndex {
 			return fmt.Errorf("attempted to restore an "+
 				"unsigned remote update: log_index=%v",
-				payDesc.LogIndex)
+				logIdx)
+		}
+
+		// We previously restored Adds along with all the other upates,
+		// but this Add restoration was a no-op as every single one of
+		// these Adds was already restored since they're all incoming
+		// htlcs on the local commitment.
+		if payDesc.EntryType == Add {
+			continue
+		}
+
+		var (
+			height    uint64
+			heightSet bool
+		)
+
+		// If pendingRemoteCommit is non-nil, we should check whether
+		// any of these updates are included in the commitment.
+		if pendingRemoteCommit != nil {
+			if logIdx < pendingRemoteCommit.theirMessageIndex {
+				height = pendingRemoteCommit.height
+				heightSet = true
+			}
 		}
 
 		// Insert the update into the log. The log update index doesn't
 		// need to be incremented (hence the restore calls), because its
 		// final value was properly persisted with the last local
 		// commitment update.
 		switch payDesc.EntryType {
-		case Add:
-			lc.remoteUpdateLog.restoreHtlc(payDesc)
-
-			// Sanity check to be sure that we are not restoring an
-			// add update that the remote hasn't signed for yet.
-			if payDesc.HtlcIndex >= lc.remoteUpdateLog.htlcCounter {
-				return fmt.Errorf("attempted to restore an "+
-					"unsigned remote htlc: htlc_index=%v",
-					payDesc.HtlcIndex)
+		case FeeUpdate:
+			if heightSet {
```

Comment:
Since the zero commitment (the very first one) can't (atm) have any updates applied, we could instead rely on this value being zero (the `height`). On the other hand, there's some merit in having this explicit variable as well....

Reply:
I prefer the explicit variable, harder to mess up in the future.